modules/vmfcore/include/vmf/ireader.hpp

```diff
@@ -64,6 +75,24 @@ class VMF_EXPORT IReader
         std::vector<std::shared_ptr<MetadataInternal>>& metadata ) = 0;
 
     virtual bool parseVideoSegments(const std::string& text, std::vector<std::shared_ptr<MetadataStream::VideoSegment> >& segments) = 0;
+
+protected:
```

Comment:
1st variant implemented
I'm still not sure if there should be a separate file for ReaderBase class